raster/src/main/scala/geotrellis/raster/io/geotiff/ArraySegmentBytes.scala

```diff
@@ -45,6 +41,10 @@ class ArraySegmentBytes(compressedBytes: Array[Array[Byte]]) extends SegmentByte
     * @return    An Array[Byte] that contains the bytes of the segment
     */
   def getSegment(i: Int) = compressedBytes(i)
```

Comment:
Use of `getSegment` needs to be minimized wherever possible in favor of `getSegments`. This will make all the difference in working with streaming segment sources.